### PR TITLE
Sanity checks for MergeTreeSettings

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add sanity check for MergeTree settings. If the settings are incorrect, the server will refuse to start or to create a table, printing detailed explanation to the user.

Detailed description / Documentation draft:
Follow-up for #12728.

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -147,6 +147,10 @@ MergeTreeData::MergeTreeData(
     if (relative_data_path.empty())
         throw Exception("MergeTree storages require data path", ErrorCodes::INCORRECT_FILE_NAME);
 
+    /// Check sanity of MergeTreeSettings. Only when table is created.
+    if (!attach)
```
Comment (alesapin) [inline src/Storages/MergeTree/MergeTreeData.cpp:151]: But what about ALTER SETTINGS? https://github.com/clickhouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreeData.cpp#L1653-L1655

Comment (alexey-milovidov) [reply src/Storages/MergeTree/MergeTreeData.cpp:151]: Yes, we should add sanity check there also...

**Comment by alexey-milovidov** [comment]
> AST-based query fuzzer — Logical error: 'Unknown numeric column of type: DB::ColumnNullable'. 

Fixed in #13213

**Review by alesapin — APPROVED** [review]
Wait for integration tests.

**Comment by alexey-milovidov** [comment]
@alesapin It looks like we have catched two flaky tests in this PR at once?

**Pull request merged**